third_party/move/move-core/types/src/resolver.rs

```diff
@@ -41,14 +45,7 @@ pub trait ResourceResolver {
         address: &AccountAddress,
         typ: &StructTag,
         metadata: &[Metadata],
-    ) -> Result<Option<(Vec<u8>, u64)>, Error>;
-}
-
-pub fn resource_add_cost(buf: Option<Vec<u8>>, extra: u64) -> Option<(Vec<u8>, u64)> {
-    buf.map(|b| {
-        let len = b.len() as u64 + extra;
-        (b, len)
-    })
+    ) -> Result<(Option<Vec<u8>>, usize), Error>;
```

Comment:
I would try to avoid changes in those apis. Given that we expect the Move repo to be used by other entities, changes like this might be breaking for them.

Reply:
Good point. Although I'm not aware of a good way to support the new semantics (bytes loaded could differ from the data read) and keep the old interface at the same time. 
I feel like what we can do in the future, is to evolve this into a clearer & more generic interface and try to upstream it.

Reply:
Anyway, let me think a little bit more about this and see if I can come up with something.